### PR TITLE
Refine remote fs doc

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of docs revamp #7666
Remote FS doc already looks quite updated to me.. Just made some small changes and added its reference to checkpointing and logging doc. Let me know if there's any additional info we should add here.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/common/checkpointing.rst [hunk 1]
```diff
@@ -385,3 +385,14 @@ Custom Checkpoint IO Plugin
...
+-----------
+
+***************************
+Managing Remote Filesystems
+***************************
+
+Lightning supports saving and loading checkpoints from a variety of filesystems, including local filesystems and several cloud storage providers
+such as ``s3`` on AWS, ``gcs`` on Google Cloud, or ``adl`` on Azure.
+
+Checkout :ref:`remote_fs` doc for more info.
```
Comment (awaelchli) [inline docs/source/common/checkpointing.rst:398]: checkout != check out

Comment (awaelchli) [inline docs/source/common/checkpointing.rst:398]:
```suggestion
Check out the :ref:`remote_fs` documentation for more info.
```

note, the displayed text here would change if the title in the doc changes too, I think. Better to put a custom display text imo

File: docs/source/extensions/logging.rst [hunk 2]
```diff
@@ -388,3 +391,14 @@ in the `hparams tab <https://pytorch.org/docs/stable/tensorboard.html#torch.util
...
+-----------
+
+***************************
+Managing Remote Filesystems
+***************************
+
+Lightning supports saving logs to a variety of filesystems, including local filesystems and several cloud storage providers
+such as ``s3`` on AWS, ``gcs`` on Google Cloud, or ``adl`` on Azure.
+
+Checkout :ref:`remote_fs` doc for more info.
```
Comment (awaelchli) [inline docs/source/extensions/logging.rst:404]: same

**Review by awaelchli — APPROVED** [review]

File: docs/source/common/checkpointing.rst (hunk 1 above)
Comment (rohitgr7) [reply docs/source/common/checkpointing.rst:398]: good point!

File: docs/source/common/checkpointing.rst [hunk 3]
```diff
@@ -385,3 +385,14 @@ Custom Checkpoint IO Plugin
...
     Some ``TrainingTypePlugins`` like ``DeepSpeedStrategy`` do not support custom ``CheckpointIO`` as checkpointing logic is not modifiable.
+
+-----------
+
+***************************
+Managing Remote Filesystems
+***************************
+
+Lightning supports saving and loading checkpoints from a variety of filesystems, including local filesystems and several cloud storage providers
+such as ``s3`` on AWS, ``gcs`` on Google Cloud, or ``adl`` on Azure.
```
Comment (Borda) [inline docs/source/common/checkpointing.rst:396]: can we add links to read more?

**Review by Borda — APPROVED** [review]

File: docs/source/common/checkpointing.rst (hunk 3 above)
Comment (rohitgr7) [reply docs/source/common/checkpointing.rst:396]: done!

**Review by ethanwharris — APPROVED** [review]

**Pull request merged**